pennylane/beta/tapes/qubit_param_shift.py

```diff
@@ -0,0 +1,220 @@
+            idx (int): trainable parameter index to differentiate with respect to
+            device (.Device, .QubitDevice): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): the quantum tape operation parameters
+
+        Returns:
+            array[float]: 1-dimensional array of length determined by the tape output
+            measurement statistics
+        """
+        # Temporarily convert all variance measurements on the tape into expectation values
+        for i in self.var_idx:
+            self._measurements[i].return_type = qml.operation.Expectation
+
+        # Get <A>, the expectation value of the tape with unshifted parameters. This is only
+        # calculated once, if `self._evA` is not None.
+        if self._evA is None:
+            self._evA = np.asarray(self.execute_device(params, device))
+
+        # evaluate the analytic derivative of <A>
+        pdA = self.parameter_shift(idx, device, params, **options)
+
+        # For involutory observables (A^2 = I) we have d<A^2>/dp = 0.
+        # Currently, the only observable we have in PL that may be non-involutory is qml.Hermitian
+        involutory = [i for i in self.var_idx if self.observables[i].name != "Hermitian"]
+
+        # If there are non-involutory observables A present, we must compute d<A^2>/dp.
+        non_involutory = set(self.var_idx) - set(involutory)
+        original = []
+
+        for i in non_involutory:
+            # We need to calculate d<A^2>/dp; to do so, we replace the
+            # involutory observables A in the queue with A^2.
+            original[:0] = [self._measurements[i]]
+            obs = self._measurements[i].obs
+
+            w = obs.wires
+            A = obs.matrix
+
+            new_obs = qml.Hermitian(A @ A, wires=w)
+            new_obs.return_type = qml.operation.Expectation
+
+            new_measurement = MeasurementProcess(qml.operation.Expectation, obs=new_obs)
+            self._measurements[i] = new_measurement
+
+        pdA2 = 0
+
+        if non_involutory:
+            # Non-involutory observables are present; the partial derivative of <A^2>
+            # may be non-zero. Here, we calculate the analytic derivatives of the <A^2>
+            # observables.
+            pdA2 = self.parameter_shift(idx, device, params, **options)
+
+            if involutory:
+                # We need to explicitly specify that the gradient of
+                # the involutory observables is 0, since we saved on processing
+                # by not replacing these observables with their square.
+                pdA2[np.array(involutory)] = 0
+
+        # restore the original observables
+        self._measurements = self._original_measurements
+
+        for i in non_involutory:
+            self._measurements[i] = original.pop()
+
+        for i in self.var_idx:
+            self._measurements[i].return_type = qml.operation.Variance

```

Comment:
Another case where we are relying on the operation annotation behaviour that we were trying to get away from.